### PR TITLE
🤑 money

### PR DESCRIPTION
**Pull request by adriancasares**

### CODE REVIEW

**Comment by reesericci** [comment]
doing code review now

**Review by reesericci — CHANGES_REQUESTED** [review]
Please fix everything to use ES modules. (Puppeteer)

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,759 @@
...
+
+  let schools: [string, SchoolType][] = [];
+
+  if (schoolType === "High") schools = highSchools;
+  else if (schoolType === "Middle") schools = middleSchools;
+  else {
+    schools = highSchools.concat(middleSchools);
+  }
+
+  const promises: Promise<School>[] = schools.map((school) => {
```
Comment (reesericci) [inline src/index.ts:111]: what is this for?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,759 @@
...
+  GetTeam,
+  GetTeams,
+  Player,
+  School,
+  SchoolType,
+  Sport,
+  Team,
+} from "./scraper";
+
+import puppeteer = require("puppeteer");
```
Comment (reesericci) [inline src/index.ts:24]: aformentioned  esmodule

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,759 @@
...
+            name: asSchoolType.name,
+            schoolType: school[1],
+            sports: asSchoolType.sports,
+          });
+        }
+      );
+    });
+  });
+
+  return await Promise.all(promises);
```
Comment (reesericci) [inline src/index.ts:127]: promise.all?

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "commonjs",
```
Comment (reesericci) [inline tsconfig.json:4]: esmodules?

**Review by reesericci — COMMENTED** [review]
Great code overall though, I will give it a shot in my editor in a sec!

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,759 @@
...
+        D: districtId,
+        S: schoolId,
+        Mt: "0",
+      },
+    }
+  );
+
+  const dom = parse(school.data);
+
+  const schoolName = dom.querySelector("#lbl_PageTitle")?.innerText ?? "";
```
Comment (reesericci) [inline src/index.ts:202]: would null be more appropriate here rather than a blank string?

File: src/index.ts (hunk 1 above)
Comment (reesericci) [reply src/index.ts:111]: the promises const

Comment (adriancasares) [reply src/index.ts:111]: Lines 77-109 gather School IDs by parsing the HTML, the promises fetch data about those schools so the function can return an array of `School`s, not just school ids.

File: src/index.ts (hunk 3 above)
Comment (adriancasares) [reply src/index.ts:127]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

File: src/index.ts (hunk 5 above)
Comment (adriancasares) [reply src/index.ts:202]: the types you made don't allow `null`

Comment (reesericci) [reply src/index.ts:202]: oh, would it be worth fixing that, or leaving it a string

File: src/index.ts (hunk 1 above)
Comment (reesericci) [reply src/index.ts:111]: ohhh that makes sense

File: src/index.ts (hunk 3 above)
Comment (reesericci) [reply src/index.ts:127]: addressed in my other comment about promises const. I know how promise.all works

File: src/index.ts (hunk 5 above)
Comment (adriancasares) [reply src/index.ts:202]: thats what I did, it's left as `""`

File: src/index.ts (hunk 2 above)
Comment (reesericci) [reply src/index.ts:24]: apparently its a puppeteer problem

File: src/index.ts (hunk 5 above)
Comment (reesericci) [reply src/index.ts:202]: resolved on discord

**Review by reesericci — APPROVED** [review]
Looks good! Will merge and payout soon.

**Pull request merged**